src/components/MenuPanel/MenuPanelAppsLoader.js

```diff
@@ -1,78 +1,134 @@
+        native
       >
-        {({ prepareProgress, hideProgress }) => (
+        {({ afterLoadingMessageProgress, showAppsProgress }) => (
           <Main>
-            <Loader>
-              <LoaderBackground style={{ opacity: prepareProgress }} />
-              <LoaderContent>
-                <LoadingRingWrapper>
-                  <LoadingRing spin={loading} />
-                </LoadingRingWrapper>
-                {loading ? 'Loading apps…' : 'Apps loaded.'}
-              </LoaderContent>
-            </Loader>
+            <Status
+              style={{
+                cursor:
+                  appsStatus === APPS_STATUS_ERROR ? 'pointer' : 'inherit',
+              }}
+              onClick={
+                appsStatus === APPS_STATUS_ERROR ? this.props.onRetry : noop
+              }
+            >
+              <StatusBackground
+                style={{ opacity: afterLoadingMessageProgress }}
+              />
+              <StatusContent>
+                <StatusIndicatorWrapper
+                  vAlign={appsStatus !== APPS_STATUS_ERROR}
+                >
+                  {appsStatus === APPS_STATUS_ERROR ? (
+                    <IconAttention />
+                  ) : (
+                    <LoadingRing spin={appsStatus === APPS_STATUS_LOADING} />
+                  )}
+                </StatusIndicatorWrapper>
+                {(() => {
+                  if (appsStatus === APPS_STATUS_LOADING) {
+                    return 'Loading apps…'
+                  }
+                  if (appsStatus === APPS_STATUS_READY) {
+                    return 'Apps loaded.'
+                  }
+                  if (appsStatus === APPS_STATUS_ERROR) {
+                    return (
+                      <div>
+                        <div>Apps loading error</div>
+                        <Text size="xxsmall">Click to try again.</Text>
+                      </div>
+                    )
+                  }
+                  return null
+                })()}
+              </StatusContent>
+            </Status>
             <Apps
               style={{
-                height: loadingTransitionDone
-                  ? 'auto'
-                  : `${lerp(prepareProgress, 40, itemsCount * 40)}px`,
-                transform: `translateX(-${(1 - hideProgress) * 100}%)`,
+                height: afterLoadingMessageProgress.interpolate(v => {
+                  const height = lerp(
+                    Math.min(1, v),
+                    40,
+                    (appsStatus === APPS_STATUS_READY ? itemsCount : 2) * 40
+                  )
+                  return `${height}px`
+                }),
+                transform: showAppsProgress.interpolate(
+                  v => `
+                    translate3d(-${(1 - Math.min(1, v)) * 100}%, 0, 0)
+                  `
+                ),
               }}
             >
-              {children(loadingTransitionDone)}
+              {children(false)}
```

Comment:
Should this return true at some point? Otherwise I think multiple app instances in a group will stay collapsed?

Reply:
Totally forgot to remove that, thanks!